log4j: java.lang.IllegalStateException: Maximum number of attachments exceeded

### Describe the bug

adding dependency `log4j-jboss-logmanager` to the application leads to the following exception when executing the integration test:

```
2022-06-28 11:00:12,440 ERROR [io.qua.ver.htt.run.QuarkusErrorHandler] (executor-thread-0) HTTP Request to /foo failed, err
or id: 9985f0d2-3c6f-4b03-938f-c76130e6048b-1: java.lang.ExceptionInInitializerError
...
Caused by: java.lang.IllegalStateException: Maximum number of attachments exceeded
        at org.jboss.logmanager.LoggerNode.attachIfAbsent(LoggerNode.java:485)
        at org.jboss.logmanager.Logger.attachIfAbsent(Logger.java:196)
        at org.apache.log4j.JBossLogManagerFacade$2.run(JBossLogManagerFacade.java:102)
        at org.apache.log4j.JBossLogManagerFacade$2.run(JBossLogManagerFacade.java:85)
        at org.apache.log4j.JBossLogManagerFacade.doPrivileged(JBossLogManagerFacade.java:269)
        at org.apache.log4j.JBossLogManagerFacade.getLoggerRepository(JBossLogManagerFacade.java:85)
        at org.apache.log4j.JBossLogManagerFacade.getLoggerRepository(JBossLogManagerFacade.java:73)
        at org.apache.log4j.LogManager.getLoggerRepository(LogManager.java:49)
        at org.apache.log4j.LogManager.getLogger(LogManager.java:57)
        at org.apache.log4j.Logger.getLogger(Logger.java:39)
        at com.lodh.foo.Bar.<clinit>(Bar.java:14)
        ... 33 more
```

Surprisingly it works when I compile the application with `skipTests`, and I execute the `java -jar` command. It only fails when the application is started through the integration tests.

### Expected behavior

log4j logging working.

### Actual behavior

It fails during the tests with an exception.

### How to Reproduce?

Not sure at this point. This is happening on the integration test application that validates an extension with multiple dependencies.
I have tried to go back to `2.9.2`, and this fails also. So it is not a regression in `2.10.0`
Would you be able to suggest some things I could test in my build (beside trying to create a full reproducer)?

### Output of `uname -a` or `ver`

_No response_

### Output of `java -version`

_No response_

### GraalVM version (if different from Java)

_No response_

### Quarkus version or git rev

2.10.0.Final

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

_No response_

### Additional information

_No response_

this seems to be the same topic as: https://github.com/quarkusio/quarkus/issues/12659 and https://github.com/jboss-logging/slf4j-jboss-logmanager/pull/10

I have rebuilt `org.jboss.logmanager:jboss-logmanager-embedded:jar:1.1.0` with some additional information:

```
s = "==> key=" + key + " value=" + value + " att1Key=" + att1Key + " att2Key=" + att2Key + " att3Key=" + att3Key;
throw new IllegalStateException("Maximum number of attachments exceeded ==> " + s);
```

how come I have 2 different `org.apache.log4j.spi.RootLogger`, and 2 `org.apache.log4j.Hierarchy`?

I checked the dependency tree only has `org.jboss.logmanager:log4j-jboss-logmanager:jar:1.3.0.Final:compile` with respect to log4j.

here are the other embedded logging libraries inside the quarkus test app: 
```
[INFO] |  |  +- org.jboss.logging:commons-logging-jboss-logging:jar:1.0.0.Final:compile
[INFO] |  |  \- org.jboss.logging:jboss-logging:jar:3.5.0.Final:compile
...
[INFO] |     +- org.jboss.logging:jboss-logging-annotations:jar:2.2.1.Final:compile
[INFO] |     +- org.slf4j:slf4j-api:jar:1.7.36:compile
[INFO] |     +- org.jboss.slf4j:slf4j-jboss-logmanager:jar:1.1.0.Final:compile
```

during the test execution, I get:

```
        Caused by: java.lang.IllegalStateException: Maximum number of attachments exceeded ==>
 key=5 value=org.apache.log4j.Hierarchy@405ac683
 att1Key=0 att1=org.apache.log4j.spi.RootLogger@3e59d5af
 att2Key=1 att2=org.apache.log4j.Hierarchy@5c74bb78
 att3Key=4 att3=org.apache.log4j.spi.RootLogger@56c4ce99
```

> Do you use a combo of different logging facades like log4j, slf4j, log4j2?

- log4j2 no I do not think so
- log4j yes (dependent lib I don't have an immediate access to)
- slf4j yes
- jboss logging yes
- apache commons logging no

--- a/jakarta/rewrite.yml
+++ b/jakarta/rewrite.yml
@@ -339,7 +339,7 @@ recipeList:
       addIfMissing: true
   - org.openrewrite.maven.ChangePropertyValue:
       key: jboss-logmanager.version
-      newValue: 1.1.0
+      newValue: 1.1.1
 ---
 type: specs.openrewrite.org/v1beta/recipe
 name: io.quarkus.jakarta-json-switch

